src/main/scala/intellij/haskell/intention/RemoveParensIntention.scala

```diff
@@ -0,0 +1,25 @@
+package intellij.haskell.intention
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import intellij.haskell.psi.HaskellPsiUtil
+import intellij.haskell.psi.HaskellTypes._
+
+class RemoveParensIntention extends PsiElementBaseIntentionAction {
+  override def invoke(project: Project, editor: Editor, psiElement: PsiElement): Unit = {
+    val (start, end) = HaskellPsiUtil.getSelectionStartEnd(psiElement, editor)
+    start.delete()
+    end.delete()
+  }
+
+  override def isAvailable(project: Project, editor: Editor, psiElement: PsiElement): Boolean = {
+    val (start, end) = HaskellPsiUtil.getSelectionStartEnd(psiElement, editor)
+    psiElement.isWritable && start.getNode.getElementType == HS_LEFT_PAREN && end.getNode.getElementType == HS_RIGHT_PAREN
```

Comment:
Because in `invoke` we just remove `start` and `end` elements, so this intention should only work when you select an expression which begins with left paren and ends with right paren.